### PR TITLE
On Ubuntu 16.04 __x64_sys_read symbol is sys_read

### PR DESCRIPTION
**Pull request by JoeyJiao**

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -3,7 +3,7 @@ override M = $(PWD)
 endif
 
 ifndef LINUX_SYSTEM_MAP
-override LINUX_SYSTEM_MAP = /boot/System.map-$(shell uname -r)
+override LINUX_SYSTEM_MAP = /proc/kallsyms
```
Comment (vanhauser-thc) [inline src/Makefile:6]: not all might have kallsyms available, so have the System.map as a fall back

File: src/threads.c [hunk 2]
```diff
@@ -38,7 +38,9 @@ static struct task_struct *next_tid(struct task_struct *start) {
 
 void recover_threads_snapshot(struct task_data *data) {
 
-  struct task_struct *t = get_task_struct(data->tsk->group_leader);
+  struct task_struct *t = data->tsk->group_leader;
+  atomic_inc(&(data->tsk->group_leader)->usage);
+
```
Comment (vanhauser-thc) [inline src/threads.c:43]: this change makes 5.x kernels fail.
you have to add an #if for the kernel version and find the right cut-off point :)

Comment (hexcoder-) [reply src/threads.c:43]: `#include <linux/version.h>
`
...
```
void recover_threads_snapshot(struct task_data *data) {

#if LINUX_VERSION_CODE < KERNEL_VERSION(5,1,0)
  struct task_struct *t = data->tsk->group_leader;
  atomic_inc(&t->usage);
#elif LINUX_VERSION_CODE < KERNEL_VERSION(5,4,0)
  struct task_struct *t = data->tsk->group_leader;
  refcount_inc(&t->usage);
#else
  struct task_struct *t = get_task_struct(data->tsk->group_leader);
#endif

```
should be like this.

Comment (hexcoder-) [reply src/threads.c:43]: even better, works universally (without `#ifdef`s) with these lines instead:
```
  struct task_struct *t = data->tsk->group_leader;
  get_task_struct(t);
```
Checked with Ubuntu 18.04 (kernel 4.15.0), Arch Linux (kernel 5.8.7)

**Comment by vanhauser-thc** [comment]
I created a new branch to test this. thanks!

**Comment by vanhauser-thc** [comment]
@JoeyJiao  @andreafioraldi I made some enhancements and fixes, is fine now IMHO

**Pull request merged**